### PR TITLE
Questions form fix

### PR DESCRIPTION
**Pull request by MySmilingTurban**
Added functionality to create new word whenever a new word option is added inside a question.
Also, adds a restriction that questions with sentences as its options shall be of type meaning.

### CODE REVIEW

File: shabadkosh/src/components/words/AddWord.tsx [hunk 1]
```diff
@@ -448,25 +409,45 @@ const AddWord = () => {
...
-
-        // make list of docRefs from selectedWordlists
-        formData.lWordlists = selectedWordlists.map((docu) => docu.id);
-        addNewWord(formData);
+    const wordIsNew = await isWordNew(formData.word);
+    if (!wordIsNew) {
+      alert('Word already exists!');
+      setIsLoading(false);
+    } else if (!validOptions) {
+      alert('Only question type - meaning can have sentences as options!');
```
Comment (theamanjs) [inline shabadkosh/src/components/words/AddWord.tsx:419]: strings should come from i18n

File: shabadkosh/src/components/words/AddWord.tsx [hunk 2]
```diff
@@ -448,25 +409,45 @@ const AddWord = () => {
...
+            formData.lQuestions = qData;
+
+            formData.part_of_speech = formData.part_of_speech ?? PARTS_OF_SPEECH.NOUN;
+            let lStatus;
+            if (type === 'create') {
+              lStatus = STATUS.CREATING_ENGLISH;
+            } else if (type === 'review') {
+              lStatus = STATUS.REVIEW_ENGLISH;
+            }
+            formData.status = lStatus ?? STATUS.CREATING_ENGLISH;
```
Comment (theamanjs) [inline shabadkosh/src/components/words/AddWord.tsx:442]: rename the lStatus to a better name which is obvious

File: shabadkosh/src/components/words/EditWord.tsx [hunk 3]
```diff
@@ -470,7 +472,7 @@ const EditWord = () => {
 
   const editWord = async (formData: any) => {
     setIsLoading(true);
-    const { fSentences, fQuestions, ...form } = formData;
+    const { lSentences, lQuestions, ...form } = formData;
```
Comment (theamanjs) [inline shabadkosh/src/components/words/EditWord.tsx:475]: What are lSentence and lQuestions? Please use elaborative names.

File: shabadkosh/src/components/words/EditWord.tsx [hunk 4]
```diff
@@ -623,29 +558,57 @@ const EditWord = () => {
...
+              if (type === 'create') {
+                lStatus = STATUS.CREATING_ENGLISH;
+              } else if (type === 'review') {
+                if ([STATUS.CREATING_ENGLISH, STATUS.FEEDBACK_ENGLISH].includes(word.status)) {
+                  lStatus = STATUS.REVIEW_ENGLISH;
+                } else if ([STATUS.CREATING_PUNJABI, STATUS.FEEDBACK_PUNJABI].includes(word.status)) {
+                  lStatus = STATUS.REVIEW_FINAL;
+                }
+              } else if (type === 'approve') {
+                if (word.status === STATUS.REVIEW_ENGLISH) {
```
Comment (theamanjs) [inline shabadkosh/src/components/words/EditWord.tsx:595]: This logic is somewhat similar in addword file. Make a common function that can serve for both the places.

**Review by theamanjs — CHANGES_REQUESTED** [review]

File: shabadkosh/src/components/words/EditWord.tsx (hunk 3 above)
Comment (MySmilingTurban) [reply shabadkosh/src/components/words/EditWord.tsx:475]: Veerji, those are local scope variables for sentences and questions, I have now changed them to sentences, questions.
We are using the same variable names for managing their state with useState, that's why added "l" in the beginning
Lmk, what would be a better solution

**Review by lbk003 — COMMENTED** [review]
lgtm

**Review by Gauravjeetsingh — CHANGES_REQUESTED** [review]

**Review by Gauravjeetsingh — CHANGES_REQUESTED** [review]

File: shabadkosh/src/components/util/utils.ts [hunk 5]
```diff
@@ -104,25 +163,29 @@ export const convertTimestampToDate = (timestamp: TimestampType, t: any) => {
...
-export const compareUpdatedAt = (a: NewWordType, b: NewWordType) => {
-  if (a.updated_at < b.updated_at) {
+export const compareUpdatedAt = (word1: any, word2: any) => {
+  if (word1.updated_at < word2.updated_at) {
     return 1;
-  } if (a.updated_at > b.updated_at) {
+  } if (word1.updated_at > word2.updated_at) {
     return -1;
   }
   return 0;
```
[the remark this answers is not in the document]
Comment (MySmilingTurban) [reply shabadkosh/src/components/util/utils.ts:191]: updated_at has Firebase Timestamp datatype which looks like the following:

```
export interface TimestampType {
  seconds: number,
  nanoseconds: number,
}
```

**Comment by MySmilingTurban** [comment]
@Gauravjeetsingh Veerji

**Review by amitojsingh — APPROVED** [review]

**Comment by amitojsingh** [comment]
Looks good to me. Good job veer ji.

**Pull request merged**